packages/manager/src/mocks/presets/crud/handlers/vpcs.ts

```diff
@@ -0,0 +1,437 @@
+      const id = Number(params.id);
+      const vpc = await mswDB.get('vpcs', id);
+
+      if (!vpc) {
+        return makeNotFoundResponse();
+      }
+
+      const deleteSubnetPromises = [];
+
+      for (const subnet of vpc.subnets) {
+        deleteSubnetPromises.push(
+          mswDB.delete('subnets', subnet.id, mockState)
+        );
+      }
+
+      await Promise.all(deleteSubnetPromises);
+      await mswDB.delete('vpcs', id, mockState);
+
+      queueEvents({
+        event: {
+          action: 'vpc_delete',
+          entity: {
+            id: vpc.id,
+            label: vpc.label,
+            type: 'vpc',
+            url: `/v4/vpc/${vpc.id}`,
+          },
+        },
+        mockState,
+        sequence: [{ status: 'notification' }],
+      });
+
+      return makeResponse({});
+    }
+  ),
+];
+
+export const createSubnet = (mockState: MockState) => [
+  http.post(
+    '*/v4beta/vpcs/:id/subnets',
+    async ({
+      params,
+      request,
+    }): Promise<StrictResponse<APIErrorResponse | Subnet>> => {
+      const vpcId = Number(params.id);
+      const vpc = await mswDB.get('vpcs', vpcId);
+
+      if (!vpc) {
+        return makeNotFoundResponse();
+      }
+
+      const payload = await request.clone().json();
+
+      const subnet = subnetFactory.build({
+        ...payload,
+        created: DateTime.now().toISO(),
+        linodes: [],
+        updated: DateTime.now().toISO(),
+      });
+
+      const updatedVPC = {
+        ...vpc,
+        subnets: [...vpc.subnets, subnet],
+      };
+
+      await mswDB.add('subnets', [vpcId, subnet], mockState);
+
+      await mswDB.update('vpcs', vpcId, updatedVPC, mockState);
+
+      queueEvents({
+        event: {
+          action: 'subnet_create',
+          entity: {
+            id: vpcId,
+            label: vpc.label,
+            type: 'subnet',
+            url: `/v4/vpcs/${vpcId}/subnets`,
+          },
+        },
+        mockState,
+        sequence: [{ status: 'notification' }],
+      });
+
+      return makeResponse(subnet);
+    }
+  ),
+];
+
+export const updateSubnet = (mockState: MockState) => [
+  http.put(
+    '*/v4beta/vpcs/:id/subnets/:subnetId',
+    async ({
+      params,
+      request,
+    }): Promise<StrictResponse<APIErrorResponse | Subnet>> => {
+      const vpcId = Number(params.id);
+      const subnetId = Number(params.subnetId);
+      const vpc = await mswDB.get('vpcs', vpcId);
+      const subnetFromDB = await mswDB.get('subnets', subnetId);
+
+      if (!vpc || !subnetFromDB) {
+        return makeNotFoundResponse();
+      }
+
+      const payload = await request.clone().json();
+
+      const updatedSubnet = {
+        ...subnetFromDB[1],
+        ...payload,
+        updated: DateTime.now().toISO(),
+      };
+
+      const updatedVPC = {
+        ...vpc,
+        subnets: vpc.subnets.map((subnet) => {
+          if (subnet.id !== subnetFromDB[1].id) {
+            return updatedSubnet;
+          }
+
+          return subnet;
```

Comment:
I believe this should be the other way round? 
```suggestion
          if (subnet.id === subnetFromDB[1].id) {
            return updatedSubnet;
          }

          return subnet;
```

Reply:
ah yes thank you!